R/bm-tpc-h.R

```diff
@@ -203,14 +203,16 @@ find_input_func <- function(func) {
 #' @param scale_factor what scale factor to reference
 #' @param query_id which query is being used
 #' @param format which format
+#' @param compression which compression to use (default: "uncompressed")
 #' @param con a connection
-#' @param memory_map should the file be memory mapped?
+#' @param memory_map should the file be memory mapped? (only relevant for the "native" format with Arrow)
```

Comment:
This confuses me.  My understanding of the "native" format is that a table is passed into the exec plan.  So why would memory mapping have any role here?

Reply:
Yeeeah, these names are probably more confusing than they need to be. FWIW, the default for `memory_map` is the one we want for arrowbench. But here it goes: 

"native" format uses `read_feather(file, as_data_frame = FALSE)` to get an Arrow table. However, by default that will use memory mapping so the first time it's called `read_feather(file, as_data_frame = FALSE)` is basically instantaneous since it's not really reading the data in, but then the first query run will be the time it takes to copy the data from disk to memory and then the duration of the query. But because of caching, the second run will be just the query. So for native formats we disable memory mapping in order to ensure we have the full table read in (_before_ we start measuring the query run time) first.

Reply:
And for our file-based formats "parquet" and "feather", we're using `open_dataset()` which preforms better since even with a single file, we can do some filtering/column reading pushdown with that as opposed to `read_parquet(...) %>% the rest of the query`

Reply:
That makes sense now.  Also, even without column/filter pushdown `open_dataset` will allow you to start processing the data before the entire table is read in (assuming the file has more than one row group of course :laughing: )

Reply:
Although I think I'd lean towards just not having `mmap` as a parameter (and always specifying false).

Reply:
That makes sense, yeah maybe I will pull it out at some point. But to keep this PR small I'm going to leave it in + merge